static/js/settings_org.js

```diff
@@ -606,6 +624,19 @@ function _set_up() {
         }
     });
 
+    $("#id_realm_org_join_restrictions").change(function (e) {
+        var org_join_restrictions = e.target.value;
+        var node = $("#allowed_domains_label").parent();
+        if (org_join_restrictions === 'only_selected_domain') {
+            node.show();
+            if (_.isEmpty(page_params.realm_domains)) {
+                overlays.open_modal('realm_domains_modal');
```

Comment:
Mysteriously this isn't working properly. When I observe DOM tree(and some console statements in overlays.js) I can see that at first, it opens the modal but immediately it get closed(and hence no modal appears). I can't figure out why this is happening :(

Reply:
Actually here I was trying to do this(see below) discussed [here](https://chat.zulip.org/#narrow/stream/101-design/subject/Join.20Org.20Dropdown/near/553913):
>Currently, If there is no selected domain and if we check the checkbox it opens the modal automatically and if we "cancel"/"add a domain" it remains unchecked(we have to again check it to enable it). So what should be this behavior for dropdown. How about opening modal when we select "restrict to list of domain" and if we cancel it reset to the original value of dropdown i.e. show this option only when there are domains in the list.

Reply:
Hmm, weird.  Let me know if you need help debugging.

Reply:
@timabbott I've tried tracking but looks like this is something which very hard to debug as it looks like triggering some event which is causing closing the modal.